src/core/debugger/gdbstub.cpp

```diff
@@ -0,0 +1,389 @@
+            SendReply(GDB_STUB_REPLY_OK);
+        } else {
+            SendReply(GDB_STUB_REPLY_ERR);
+        }
+        break;
+    }
+    case 's':
+        actions.push_back(DebuggerAction::StepThread);
+        break;
+    case 'C':
+    case 'c':
+        actions.push_back(DebuggerAction::Continue);
+        break;
+    case 'Z': {
+        const auto addr_sep{std::find(command.begin(), command.end(), ',') - command.begin() + 1};
+        const size_t addr{static_cast<size_t>(strtoll(command.data() + addr_sep, nullptr, 16))};
+
+        if (system.Memory().IsValidVirtualAddress(addr)) {
+            replaced_instructions[addr] = system.Memory().Read32(addr);
+            system.Memory().Write32(addr, arch->BreakpointInstruction());
+            system.InvalidateCpuInstructionCacheRange(addr, sizeof(u32));
+
+            SendReply(GDB_STUB_REPLY_OK);
+        } else {
+            SendReply(GDB_STUB_REPLY_ERR);
+        }
+        break;
+    }
+    case 'z': {
+        const auto addr_sep{std::find(command.begin(), command.end(), ',') - command.begin() + 1};
+        const size_t addr{static_cast<size_t>(strtoll(command.data() + addr_sep, nullptr, 16))};
+
+        if (system.Memory().IsValidVirtualAddress(addr) && replaced_instructions.contains(addr)) {
+            system.Memory().Write32(addr, replaced_instructions[addr]);
+            system.InvalidateCpuInstructionCacheRange(addr, sizeof(u32));
+            replaced_instructions.erase(addr);
+
+            SendReply(GDB_STUB_REPLY_OK);
+        } else {
+            SendReply(GDB_STUB_REPLY_ERR);
+        }
+
+        break;
+    }
+    default:
+        SendReply(GDB_STUB_REPLY_EMPTY);
+        break;
+    }
+}
+
+void GDBStub::HandleQuery(std::string_view command) {
+    if (command.starts_with("TStatus")) {
+        // no tracepoint support
+        SendReply("T0");
+    } else if (command.starts_with("Supported")) {
+        SendReply("PacketSize=4000;qXfer:features:read+;qXfer:threads:read+;qXfer:libraries:read+");
+    } else if (command.starts_with("Xfer:features:read:target.xml:")) {
+        const auto offset{command.substr(30)};
+        const auto amount{command.substr(command.find(',') + 1)};
+
+        const auto offset_val{static_cast<u64>(strtoll(offset.data(), nullptr, 16))};
+        const auto amount_val{static_cast<u64>(strtoll(amount.data(), nullptr, 16))};
+        const auto target_xml{arch->GetTargetXML()};
+
+        if (offset_val + amount_val > target_xml.size()) {
+            SendReply("l" + target_xml.substr(offset_val));
+        } else {
+            SendReply("m" + target_xml.substr(offset_val, amount_val));
+        }
+    } else if (command.starts_with("Offsets")) {
+        Loader::AppLoader::Modules modules;
+        system.GetAppLoader().ReadNSOModules(modules);
+
+        const auto main =
+            std::find_if(modules.begin(), modules.end(),
+                         [](std::pair<VAddr, std::string> key) { return key.second == "main"; });
```

Comment:
```suggestion
                         [](const std::pair<VAddr, std::string>& key) { return key.second == "main"; });
```

or just use `const auto&`, either is fine.